src/compiler/checker.ts

```diff
@@ -7852,16 +7842,18 @@ namespace ts {
                 case SyntaxKind.NeverKeyword:
                     return neverType;
                 case SyntaxKind.ObjectKeyword:
-                    return nonPrimitiveType;
+                    if (node.flags & NodeFlags.JavaScriptFile) {
+                        return anyType;

```

Comment:
`getPrimitiveTypeFromJSDocTypeReference` operates on TypeReferenceNode not on a TypeNode with SyntaxKind.Object. Actually, the "object" case there no longer applies since it `@param {object} x` is no longer parsed as a type reference.